pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -73,14 +84,25 @@ func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClien
 	}
 
 	adapter := &dataweaveTransformAdapter{
-		spell:               env.DwSpell,
-		incomingContentType: env.IncomingContentType,
-		outputContentType:   env.OutputContentType,
-		replier:             replier,
-		ceClient:            ceClient,
-		logger:              logger,
-		mt:                  mt,
-		sink:                env.Sink,
+		spellOverride: env.AllowDwSpellOverride,
+
+		replier:  replier,
+		ceClient: ceClient,
+		logger:   logger,
+		mt:       mt,
+		sink:     env.Sink,
+	}
+
+	if env.DwSpell != "" {
+		adapter.defaultSpell = &env.DwSpell
+	}
+	if env.InputContentType != "" {
+		adapter.defaultInputContentType = &env.InputContentType
+	}
+	if env.OutputContentType != "" {
+		adapter.defaultOutputContentType = &env.OutputContentType
+	} else {
+		adapter.defaultOutputContentType = ptr.String("application/json")
```

Comment:
Interesting. Didn't realize there was a knative pkg/method call for this now. Figured: `str := "application/json"; adapter.defaultOutputContentType = &str` would work.

You don't need to address this.